src/onchain.hs

```diff
@@ -92,6 +95,11 @@ mintValidate c action ctx = case action of
                                   where
                                     -- | We need to make sure that mint is also empty by checking if length of mint is -1, otherwise you could mint anything you want.
                                     checkHelper [] [] [] _ userL = userL == -1
+
+                                    -- REVIEW(SN): This is assuming inline datums at the locked txOuts. AFAIU they
+                                    -- are intended to be locked indefinitely. At the same time, only the transaction
+                                    -- which mints tokens needs to access these datums, right? Maybe that means that we can just
+                                    -- use normal datums and force their inclusion in the tx. This would be way cheaper on fees.
```

Comment:
The inline datum is here only a currency symbol. It has almost the same size as a datum hash. I'm not sure if that saves a lot, because I also have to check that the `lockOutDatum` matches the currency symbol of the minting policy itself. It may even be more expensive to get the datum from the witness set first. Or am I missing something?

Reply:
It's just a trade-off, but you're probably aware. For inline datums you need to pay for both, the transaction space (as the datum needs to be in the transaction creating the utxo) and the `utxoCostPerByte` (`4310` lovelace). A currency symbol is only 28 bytes? That would be an overhead of `120,680` lovelace per output (but it's a deposit, so not lost).